photon-lib/src/main/java/org/photonvision/PhotonPoseEstimator.java

```diff
@@ -470,6 +528,80 @@ private Optional<EstimatedRobotPose> pnpDistanceTrigSolveStrategy(PhotonPipeline
                         PoseStrategy.PNP_DISTANCE_TRIG_SOLVE));
     }
 
+    private Optional<EstimatedRobotPose> constrainedPnpStrategy(
+            PhotonPipelineResult result,
+            Optional<Matrix<N3, N3>> cameraMatrixOpt,
+            Optional<Matrix<N8, N1>> distCoeffsOpt,
+            ConstrainedSolvepnpParams meme) {
+        boolean hasCalibData = cameraMatrixOpt.isPresent() && distCoeffsOpt.isPresent();
+        // cannot run multitagPNP, use fallback strategy
+        if (!hasCalibData) {
+            return update(result, cameraMatrixOpt, distCoeffsOpt, null, this.multiTagFallbackStrategy);
+        }
+
+        // Need heading if heading fixed
+        if (!meme.headingFree && headingBuffer.getSample(result.getTimestampSeconds()).isEmpty()) {
+            return update(result, cameraMatrixOpt, distCoeffsOpt, null, this.multiTagFallbackStrategy);
+        }
+
+        Pose3d fieldToRobotSeed;
+
+        // Attempt to use multi-tag to get a pose estimate seed
+        if (result.getMultiTagResult().isPresent()) {
+            fieldToRobotSeed =
+                    new Pose3d()
```

Comment:
```suggestion
                    Pose3d.kZero
```
Super duper small nit, but might as well.